napi_async.md

```diff
@@ -0,0 +1,44 @@
+# Async Helpers
+
+Addon modules often need to leverage async helpers from libuv as part of their
+implementation.  This allows them to schedule work to be execute asynchronously
+so that their methods can return in advance of the work being completed.  This
+is important in order to allow them to avoid blocking overall execution
+of the Node.js application.
+
+N-API provides an abi stable interface for these
+supporting functions which covers the most common asynchronous use cases.
+
+N-API defines the ```napi_work``` structure which is used to manage
+asynchronous wokers. Instances are created/deleted with the following methods:
+
+```C
+NAPI_EXTERN napi_work napi_create_async_work();
+NAPI_EXTERN void napi_delete_async_work(napi_work w);
```

Comment:
Not a doc issue, but I just realized... why don't these APIs return `napi_status`?